File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cs, maxs = 0, float('-inf')
        for end in nums:
            cs += end
            maxs = max(maxs, cs)
            if cs < 0:
                cs = 0
        return maxs

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length;
        int maxSum = nums[0];
        int curSum = 0;
        for(int st=0; st<n; st++){
            //Base Case:
            if (curSum < 0) curSum = 0;
            curSum += nums[st];    
            maxSum = Math.max(maxSum, curSum);
        }return maxSum;
    }
}